support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20RevocationEndpointController.java

```diff
@@ -0,0 +1,194 @@
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.pac4j.core.context.JEEContext;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.ProfileManager;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.view.json.MappingJackson2JsonView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This is {@link OAuth20RevocationEndpointController}.
+ *
+ * @author Julien Huon
+ * @since 6.2.0
+ */
+@Slf4j
+public class OAuth20RevocationEndpointController extends BaseOAuth20Controller {
+    public OAuth20RevocationEndpointController(final OAuth20ConfigurationContext oAuthConfigurationContext) {
+        super(oAuthConfigurationContext);
+    }
+
+    /**
+     * Handle request for revocation.
+     *
+     * @param request  the request
+     * @param response the response
+     * @return the response entity
+     */
+    @PostMapping(path = '/' + OAuth20Constants.BASE_OAUTH20_URL + '/' + OAuth20Constants.REVOCATION_URL,
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    public ModelAndView handleRequest(final HttpServletRequest request,
+                                      final HttpServletResponse response) {
+        val context = new JEEContext(request, response, getOAuthConfigurationContext().getSessionStore());
+
+        if (!verifyRevocationRequest(context)) {
+            LOGGER.error("Revocation request verification failed. Request is missing required parameters");
+            return OAuth20Utils.writeError(response, OAuth20Constants.INVALID_REQUEST);
+        }
+
+        val manager = new ProfileManager<CommonProfile>(context, context.getSessionStore());
+        val clientId = OAuth20Utils.getClientIdAndClientSecret(context).getLeft();
+        val registeredService = getRegisteredServiceByClientId(clientId);
+
+        if (OAuth20Utils.doesServiceNeedAuthentication(registeredService)) {
+            if (manager.get(true).isEmpty()) {
+                LOGGER.error("Service [{}] requests authentication", clientId);
+                return OAuth20Utils.writeError(response, OAuth20Constants.ACCESS_DENIED);
+            }
+        } else {
+            val service = getOAuthConfigurationContext().getWebApplicationServiceServiceFactory().createService(registeredService.getServiceId());
+
+            val audit = AuditableContext.builder()
+                .service(service)
+                .registeredService(registeredService)
+                .build();
+
+            val accessResult = getOAuthConfigurationContext().getRegisteredServiceAccessStrategyEnforcer().execute(audit);
+            if (accessResult.isExecutionFailure()) {
+                return OAuth20Utils.writeError(response, OAuth20Constants.INVALID_REQUEST);
+            }
```

Comment:
Audit checks should always execute. Don't think you need to put these lines in an else clause.

Reply:
When the request is executed by a confidential client, the exact same audit log is provided by the OAuth20ClientIdClientSecret Authenticator.

I think we could add a RevocationTokenResponseAuditResourceResolver or something like that in a future evolution ?

Reply:
Makes sense, sure.

And yes, sure; `RevocationTokenResponseAuditResourceResolver` would be a good idea in the next PR.